kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/settings/SnapshotGenericModal.java

```diff
@@ -0,0 +1,333 @@
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ * 
+ * Contributors:
+ *  Eurotech
+ *******************************************************************************/
+package org.eclipse.kura.web.client.ui.settings;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.eclipse.kura.web.client.messages.Messages;
+import org.gwtbootstrap3.client.ui.Anchor;
+import org.gwtbootstrap3.client.ui.CheckBox;
+import org.gwtbootstrap3.client.ui.Form;
+import org.gwtbootstrap3.client.ui.FormLabel;
+import org.gwtbootstrap3.client.ui.Modal;
+import org.gwtbootstrap3.client.ui.ModalFooter;
+import org.gwtbootstrap3.client.ui.TextBox;
+import org.gwtbootstrap3.client.ui.html.Paragraph;
+
+import com.google.gwt.core.client.GWT;
+import com.google.gwt.event.dom.client.ClickEvent;
+import com.google.gwt.event.dom.client.KeyUpEvent;
+import com.google.gwt.event.shared.HandlerRegistration;
+import com.google.gwt.uibinder.client.UiBinder;
+import com.google.gwt.uibinder.client.UiField;
+import com.google.gwt.user.client.ui.Composite;
+import com.google.gwt.user.client.ui.Label;
+import com.google.gwt.user.client.ui.ScrollPanel;
+import com.google.gwt.user.client.ui.VerticalPanel;
+import com.google.gwt.user.client.ui.Widget;
+
+public abstract class SnapshotGenericModal extends Composite {
+
+    protected static final Messages MSGS = GWT.create(Messages.class);
+    private static SnapshotGenericModalUiBinder uiBinder = GWT.create(SnapshotGenericModalUiBinder.class);
+
+    interface SnapshotGenericModalUiBinder extends UiBinder<Widget, SnapshotGenericModal> {
+    }
+
+    @UiField
+    Modal snapshotModal;
+    @UiField
+    Form snapshotForm;
+    @UiField
+    Paragraph snapshotModalDescription;
+    @UiField
+    Paragraph snapshotModalHint;
+    @UiField
+    Label searchBoxSeparatorSmall;
+    @UiField
+    TextBox pidSearch;
+    @UiField
+    Label searchBoxSeparatorBig;
+    @UiField
+    ScrollPanel pidSelectionScrollPanel;
+    @UiField
+    Anchor selectOrRemoveAllAnchor;
+    @UiField
+    FormLabel noPidSelectedError;
+    @UiField
+    Label selectedPidCounter;
+    @UiField
+    ModalFooter snapshotFooter;
+
+    HandlerRegistration anchorClickHandler;
+
+    VerticalPanel pidPanel = new VerticalPanel();
+
+    protected SnapshotGenericModal() {
+        initWidget(uiBinder.createAndBindUi(this));
+        this.noPidSelectedError.setVisible(false);
+    }
+
+    /*
+     * Abstract initializers
+     */
+
+    /**
+     * Initialize the modal title, the description and the hint paragraphs of the modal.
+     */
+    abstract void initTitleAndDescriptionAndHint();
+
+    /**
+     * Initialize eventual buttons to perform the desired operations on the PIDs list.
+     * 
+     * @param snapshotId
+     *            - snapshotId of the snapshot from which the PIDs list was taken as {@link Long}
+     */
+    abstract void initEventButtons(Long snapshotId);
```

Comment:
`initEventButtons `-> `addButton(SnapshotSelectorActionButton button)` and define a new `SnapshotSelectorActionButton` for this purpose. Button disposal can be handled in this class